test/builtin_dataset_mocks.py

```diff
@@ -911,6 +911,61 @@ def country211(info, root, config):
     return num_examples * len(classes)
 
 
+@register_mock
+def food101(info, root, config):
+    data_folder = root / "food101"
+
+    num_images_per_class = 3
+    image_folder = data_folder / "images"
+    categories = ["apple_pie", "baby_back_ribs", "baklava"]
+    image_ids_per_category = {
+        category: [
+            str(path.relative_to(path.parents[1]).as_posix())
+            for path in create_image_folder(
+                image_folder,
+                category,
+                file_name_fn=lambda idx: f"{idx:04d}.jpg",
+                num_examples=num_images_per_class,
+            )
+        ]
+        for category in categories
+    }
+
+    # Remove the extension from the list of images per category.
+    image_ids_per_category = {k: [e.replace(".jpg", "") for e in v] for k, v in image_ids_per_category.items()}
+
+    meta_folder = data_folder / "meta"
+    meta_folder.mkdir()
+
+    with open(meta_folder / "classes.txt", "w") as file:
+        for category in categories:
+            file.write(f"{category}\n")
+
+    train_categories = categories[:2]
+    test_categories = categories[2:]
+
+    with open(meta_folder / "train.txt", "w") as file:
+        for category in train_categories:
+            for img_id in image_ids_per_category[category]:
+                # Notice that the img_id is of the form category/img_id here.
+                file.write(f"{img_id}\n")
+
+    with open(meta_folder / "test.txt", "w") as file:
+        for category in test_categories:
+            for img_id in image_ids_per_category[category]:
+                # Notice that the img_id is of the form category/img_id here.
+                file.write(f"{img_id}\n")
+
+    num_samples_map = {
+        "train": len(train_categories) * num_images_per_class,
+        "test": len(test_categories) * num_images_per_class,
+    }
```

Comment:
1. It is somewhat dangerous to split along the categories. Given that this is a fairly simple dataset, there should be no issue, but other datasets have different behavior based on the split. Thus, we should make sure that each split sees every category at least once. This is of course that this is also the case in the original data.
2. We can simplify this by using a loop over the splits.

```suggestion
    splits = ["train", "test"]
    num_samples_map = {}
    for offset, split in enumerate(splits):
        image_ids_in_split = image_ids[offset :: len(splits)]
        num_samples_map[split] = len(image_ids_in_split)

        with open(meta_folder / f"{split}.txt", "w") as file:
            for image_id in image_ids_in_split:
                file.write(f"{image_id}\n")
```